plugins/woocommerce-admin/client/tasks/fills/products/use-create-product-by-type.ts

```diff
@@ -38,7 +40,12 @@ export const useCreateProductByType = () => {
 			);
 
 			if ( assignment.variationName === 'treatment' ) {
-				navigateTo( { url: getNewPath( {}, '/add-product', {} ) } );
+				await updateOptions( {
+					[ NEW_PRODUCT_MANAGEMENT ]: 'yes',
+				} );
+				window.location.href = getAdminLink(
+					'admin.php?page=wc-admin&path=/add-product'
+				);
```

Comment:
We need this to refresh the page to use the new experience.

Reply:
That makes sense.  Do you think there's any point in checking if the feature is already enabled and navigating using `navigateTo` instead of the full page reload?

This obviously won't work the first time they click this, but if they revisit the task, it would allow navigation via React Router.

Reply:
Good call @joshuatf! I added that change in the commit [3a7015f](https://github.com/woocommerce/woocommerce/pull/36406/commits/3a7015f2ce594b05eafce24ab17dca70bf10f346).
Could you take another look at this PR?